File: ignite/problem1/n-rotation.py
# import deque object from collections
from collections import deque
def rotate_array_by_n(arr, n = 1):
  """method to rotate list by n (default n is 1)"""
  dearr = deque(arr)
  # used deque object's method `rotate` to rotate input list
  dearr.rotate(n)
  return list(dearr)

print rotate_array_by_n(arr = [1,2,3,4,5,6], n = 2)

File: ignite/problem1/n-rotation.rb
# opened Array class and added method rotation_by.
class Array
  # define: This method could be called on any array
  # and when passed with an arguement rotates array in opposite direction.
  # Without arguement rotate by 1.
  def rotation_by(n = 1)
    rotate!(n * (-1))
  end
end

arr = [1, 2, 3, 4, 5, 6]
p arr.rotation_by(2)

 # => [5, 6, 1, 2, 3, 4]